cpp/src/arrow/compute/kernels/aggregate_test.cc

```diff
@@ -1205,6 +1205,21 @@ TEST_F(TestVarStdKernelMergeStability, Basics) {
 #endif
 }
 
+// Test round-off error
+class TestVarStdKernelRoundOff : public TestPrimitiveVarStdKernel<DoubleType> {};
+
+TEST_F(TestVarStdKernelRoundOff, Basics) {
+  // build array: np.arange(321000, dtype='float64')
+  double value = 0;
+  ASSERT_OK_AND_ASSIGN(
+      auto array, ArrayFromBuilderVisitor(float64(), 321000, [&](DoubleBuilder* builder) {
+        builder->UnsafeAppend(value++);
+      }));
+
+  // reference value from numpy.var()
+  this->AssertVarStdIs(*array, VarianceOptions{0}, 8586749999.916667);
+}
```

Comment:
Should we have a similar test for `float32` as well?

Reply:
Looks not necessary, as internally `float32` is converted to `double` before calculation, and there's no `float32` summation.
Will do some tests against numpy to see if additional test is required.

Reply:
Hmm... some difference from numpy. Our variance kernel always returns `double` for any input type. Numpy can select output dtype, by default, it returns `float32` for `float32`, `double` for any other types. https://numpy.org/doc/stable/reference/generated/numpy.var.html

I would prefer always returning `double`. It's simpler and looks more reasonable.

Reply:
I agree that it looks more reasonable.

Reply:
Added float32/int32/int64 round-off test.